File: Juan/P1/S1/Napakalaki/src/napakalaki/Monster.java

package napakalaki;

/**
 *
 * @author juane
 * @version 1.0
 * 
 * Clase Monster
 */
public class Monster {
    
    /* Atributos propios de la clase*/
    
    private String name; // nombre del monstruo
    private int combatLevel; // nivel de combate del monstruo

    /* Atributo de tipo Prize y BadConsequence */
    private Prize price;
    private BadConsequence bc;
    
    /* Constructor*/
    
    Monster(String name, int level, BadConsequence bc, Prize price){
        
        this.name = name;
        this.combatLevel = level;
        this.bc = bc;
        this.price = price;
    }
    
    /**
     * Método getName()
     * Devuelve el nombre del monstruo
     * @return name de tipo int
     */
    public String getName() {
        return name;
    }

    /**
     * Método getCombatLevel()
     * Devuelve el nivel de combate del monstruo
     * @return 
     */
    public int getCombatLevel() {
        return combatLevel;
    }
    
    /**
     * Método toString
     * @return 
     */
    @Override
    public String toString() {
        return "Monster = " + name + ", combatLevel = " 
                +  combatLevel + ", Price: " +  price 
                + ", " + bc ;
    }
    
    
}

File: Juan/P2/Ruby/Napakalaki/lib/Monster.rb
#encoding: utf-8
#Versión 1.0
# author: juane, Raúl

require_relative 'BadConsequence.rb'
require_relative 'TreasureKind.rb'
require_relative  'Prize.rb'

class Monster
    #Atributos propios de la clase
  
    # @name # String para representar el nombre del monstruo
    # @combatLevel # int para representar el nivel de combate del monstruo
  
    #Otros atributos
    # @badconsequence  # de tipo BadConsequence
    # @price # de tipo Prize
 
  
  
    def initialize(name,level,bc,price)
        @name = name
        @combatLevel = level    
        @prize = price
        @badconsequence = bc
    
    end
  
     attr_reader :name
     attr_reader :combatLevel
     attr_reader :prize
     attr_reader :badconsequence
    
    

    #EXAMEN
    attr_reader :n  
    attr_reader :fav
    attr_reader :favorite # objeto de la clase Player
   
    
    def getNumOfMonster()
        
        card_d = CardDealer.new
        card_d.giveMonsterBack(m)
        tam = card_d.giveMonsterBack(m).size
        num=0
        
         card_d.each do |tam|
            if(tam !=0)
                num= num +1
            end
        end
        
        num
    end
    
    #FIN EXAMEN
    #/*Niveles ganados/
    def getLevelsGained
        @prize.levels
    end 
    
    #/*Tesoros ganados/
    def getTreasuresGained
        @prize.treasures
    end
    
    def to_s
        "Monster =  #{@name}  , combatLevel =   #{@combatLevel} ,  #{@badconsequence} ,  Price:  #{@prize} "      
    end
  
end
